src/System.Private.CoreLib/shared/System/Type.Helpers.cs

```diff
@@ -271,47 +271,47 @@ public virtual MemberInfo[] FindMembers(MemberTypes memberType, BindingFlags bin
             {
                 for (i = 0; i < m.Length; i++)
                     if (m[i] != null)
-                        ret[cnt++] = m[i];
+                        ret[cnt++] = m[i]!;
```

Comment:
Many more cases of the same question.  These all seem unnecessary ceremony and the compiler analysis shouldn't be warning here.
cc: @jaredpar

Reply:
I just got the compiler bits updated (less than 20 minutes ago), so it would likely be worthwhile rechecking some of these to see if they still repro.

Reply:
> I just got the compiler bits updated (less than 20 minutes ago), so it would likely be worthwhile rechecking some of these to see if they still repro.

Thanks.

@jaredpar, has a warning been considered where `!` is used but isn't necessary?  That would help us to remove unnecessary clutter as the compiler improves its analysis, especially in these early days.

Reply:
It's been discussed and likely doable. 

@cston

Reply:
@jaredpar @cston is there an issue open for it? I agree with @stephentoub that it would help clean our code up as you tweak the compiler.

Reply:
https://github.com/dotnet/roslyn/issues/25372